state/logs.go

```diff
@@ -542,37 +542,32 @@ func (t *logTailer) processReversed(query *mgo.Query) error {
 	// c) use the aggregation pipeline in Mongo 3.2 to write the docs to
 	//    a temp location and iterate them forward from there.
 	// (a) makes the most sense for me :)
-	if t.params.InitialLines > t.maxInitialLines {
+	if t.params.LatestLogCount > t.maxInitialLines {
 		return errors.Errorf("too many lines requested (%d) maximum is %d",
-			t.params.InitialLines, maxInitialLines)
+			t.params.LatestLogCount, maxInitialLines)
 	}
+
 	iter := query.Sort("-t", "-_id").
-		Limit(t.params.InitialLines).
+		Limit(t.params.LatestLogCount).
 		Iter()
-	defer iter.Close()
-	queue := make([]logDoc, t.params.InitialLines)
```

Comment:
Pre-allocation is preferred for efficiency. Can we adjust the logic to retain this.

Reply:
Sure will update this, thanks Ian